File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Go/s378126458_comments_removed.go
package main

import (
	"fmt"
)

func main() {
	var N, Q int
	fmt.Scan(&N, &Q)
	var s string
	fmt.Scan(&s)
	pre := make([]int, N+1)
	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}

	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}

	for Q > 0 {
		var l, r int
		fmt.Scan(&l, &r)
		r--
		fmt.Println(pre[r] - pre[l-1])
		Q--
	}
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/Python/s378126458_comments_removed.py
import sys

def main():
    input = sys.stdin.read().split()
    N = int(input[0])
    Q = int(input[1])
    s = input[2]
    pre = [0] * (N + 1)
    
    for i in range(1, N):
        if s[i - 1] == 'A' and s[i] == 'C':
            pre[i] = 1

    for i in range(1, N):
        pre[i] += pre[i - 1]

    idx = 3
    while Q > 0:
        l = int(input[idx]) - 1
        r = int(input[idx + 1]) - 1
        print(pre[r] - pre[l])
        idx += 2
        Q -= 1

if __name__ == "__main__":
    main()
# End of Code
